### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by kantai**
This is mostly updated by copying content from SIP-002.

I deleted the files that seemed to be getting updated elsewhere (namely in https://github.com/blockstack/docs.blockstack/pull/541) --

* tutorial
* sdk-quickstart

And removed some files that had documentation elsewhere or shouldn't be documented
* install-source -- this is covered in the repo README
* clarityCLI -- this CLI probably shouldn't exist anymore. all of it's features are accesible via `blockstack-cli local`, and there's another clarity cli (javascript), which is more featureful.

I also removed `functions`, since it's content was better covered in the language ref, plus the generated function docs.

### CODE REVIEW

**Comment by kantai** [comment]
I also maybe need some insight from @hstove or @agraebe on how this documentation relates to https://github.com/blockstack/docs.blockstack/pull/541

There's overlap in these files -- will deleting or editing things here create problems on the documentation site?

File: docs/smart-contract-vm.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ You should see a message saying "Database created".
 Start by type checking the "names" sample contract.
 
 ```bash
-$ blockstack-core local check sample-programs/names.clar /data/db
+$ blockstack-core local check sample-contracts/names.clar /data/db
```
Comment (jcnelson) [inline docs/smart-contract-vm.md:68]: Is `blockstack-core` still the name of the binary to run?

File: docs/smart/clarityRef.md [hunk 2]
```diff
@@ -9,141 +9,126 @@ This file contains the reference for the Clarity language.
...
-```cl
-(get-block-info time 10) ;; Returns 1557860301
+## Clarity Type System
+
+The Clarity language uses a strong static type system. Function arguments
+and database schemas require specified types, and use of types is checked
+during contract launch. The type system does _not_ have a universal
+super type. The type system contains the following types:
+
+* `(tuple (key-name-0 key-type-0) (key-name-1 key-type-1) ...)` -
```
Comment (jcnelson) [inline docs/smart/clarityRef.md:19]: Do we want to mention implicit tuple syntax?

File: docs/smart/principals.md [hunk 3]
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
-
-```
-(define (transfer! (sender principal) (recipient principal) (amount int))
-  (if (and
-        (not (eq? sender recipient))
-        (debit-balance! sender amount)
-        (credit-balance! recipient amount))
-    true
-    false))
+```cl
```
Comment (jcnelson) [inline docs/smart/principals.md:31]: scheme?

File: docs/smart/principals.md [hunk 4]
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (check-condition)
```
Comment (jcnelson) [inline docs/smart/principals.md:33]: Should `check-condition` be an explicit condition in this example?

File: docs/smart/principals.md [hunk 5]
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
 ## Smart contracts as principals
 
-Smart contracts themselves are principals and are represented by the smart contract's identifier. You create the identifier when you launch the contract, for example, the contract identifier here is `hanomine`.
+Smart contracts themselves are principals and are represented by the
+smart contract's identifier -- which is the publishing address of the
+contract _and_ the contract's name, e.g.:
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```cl
```
Comment (jcnelson) [inline docs/smart/principals.md:44]: scheme?

File: docs/smart/principals.md [hunk 6]
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
+`as-contract` function returns the value of the provided expression.
 
-For example, a smart contract that implements something like a "token faucet" could be implemented as so:
+For example, a smart contract that implements something like a "token
+faucet" could be implemented as so:
 
 ```cl
 (define-public (claim-from-faucet)
-  (if (is-none? (fetch-entry claimed-before (tuple (sender tx-sender))))
+  (if (is-none? (fetch-entry claimed-before {sender: tx-sender}))
```
Comment (jcnelson) [inline docs/smart/principals.md:64]: If we're going to use `{ }` in examples, it should be introduced earlier as the way to define tuples.

File: docs/smart/principals.md [hunk 7]
```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
...
       (let ((requester tx-sender)) ;; set a local variable requester = tx-sender
         (begin
             (insert-entry! claimed-before {sender: requester} {claimed: true})
-            (as-contract (stacks-transfer! requester 1)))))
-      (err 1))
+            (as-contract (stacks-transfer? u1 tx-sender requester))))
```
Comment (jcnelson) [inline docs/smart/principals.md:68]: Is it `stacks-transfer?` or `stx-transfer?`

File: docs/smart/clarityRef.md (hunk 2 above)
Comment (kantai) [reply docs/smart/clarityRef.md:19]: I'd prefer not to here -- I want to keep this as close as possible to the SIP and with minimal surface area for now. Once I can setup a better structure for executing automatic tests for any code in the documentation, I'd be happier to expand examples.

File: docs/smart-contract-vm.md (hunk 1 above)
Comment (kantai) [reply docs/smart-contract-vm.md:68]: Yep -- these commands are also checked in a "demo_tests" job in Circle.

File: docs/smart/principals.md (hunk 6 above)
Comment (kantai) [reply docs/smart/principals.md:64]: There's no documentation for how to declare any literals in these documents at the moment. That could be added as a section to the clarityRef.md file, I think.

File: docs/smart/principals.md (hunk 5 above)
Comment (kantai) [reply docs/smart/principals.md]: Fixed in 1a0d04d

File: docs/smart/principals.md (hunk 4 above)
Comment (kantai) [reply docs/smart/principals.md]: Added in 1a0d04d

File: docs/smart/principals.md (hunk 3 above)
Comment (kantai) [reply docs/smart/principals.md]: Fixed in 1a0d04d

File: docs/smart/principals.md (hunk 7 above)
Comment (kantai) [reply docs/smart/principals.md]: Fixed in 1a0d04d

**Comment by kantai** [comment]
Thanks @jcnelson -- I addressed the comments related to accuracy -- this PR is mostly about fixing the previously inaccurate and out-of-date details in these docs.

These files are still a pretty far way removed from comprising a full language reference, but that will be an issue for another day.

File: docs/smart/principals.md (hunk 6 above)
Comment (kantai) [reply docs/smart/principals.md]: Replaced with the explicit tuple syntax in https://github.com/blockstack/stacks-blockchain/pull/1439/commits/9eb9677a3433ce0d0949dd085ccff96b39976aec

File: docs/smart/principals.md (hunk 3 above)
Comment (njordhov) [reply docs/smart/principals.md]: I have found using triple-tick `clojure` to result in the most fitting formatting.

**Pull request merged**